Install FFMPEG with brew on Mac CI runners

GenevieveBuckley opened the pull request:
Closes https://github.com/napari/napari-animation/issues/213

This is a temporary fix for issue https://github.com/imageio/imageio-ffmpeg/issues/71, which has been causing our CI to fail.

Czaki commented on .github/workflows/test_and_deploy.yml:
```diff
@@ -32,12 +32,26 @@ jobs:
+        # https://github.com/imageio/imageio-ffmpeg/issues/71
+      - name: brew install ffmpeg
+        if: runner.os == 'macOS'
+        uses: artemnovichkov/action-homebrew@0.1.0
```
Are you sure that this action is needed? 
As I remember, the brew is preinstalled. In this action, brew is used without installation https://github.com/4DNucleome/PartSegCore-compiled-backend/blob/master/.github/workflows/wheels.yml

GenevieveBuckley commented:
The ruff CI check failed earlier because I updated ruff's target version of python from 3.8 to 3.12. To keep things simple, I've reverted that change here for now, and made a separate PR to update ruff https://github.com/napari/napari-animation/pull/218.

psobolewskiPhD commented:
Thanks for taking a look at this!

The pyside changes (at a minimum) from here:
[`23cdaa8` (#214)](https://github.com/napari/napari-animation/pull/214/commits/23cdaa8f74c6bd105a3f8e8ef863c1679bd8b13b)
will need to be reverted:
The latest pyside2 on pypi is 5.15.2.1
https://pypi.org/project/PySide2/
Which doesn't (officially) support even python 3.10, let alone 3.11 and 3.12:
https://wiki.qt.io/Qt_for_Python
<img width="543" alt="image" src="https://github.com/napari/napari-animation/assets/76622105/76cf107d-490b-4bb2-b940-8b1c0a17aa59">

Also, to echo Grzegorz's question, why not use the homebrew installed on the macOS runners?
The action you are using 
https://github.com/AnimMouse/setup-ffmpeg
Does not support arm64 as far as I can tell:
https://evermeet.cx/ffmpeg/apple-silicon-arm

The other option/alternative PR is to switch to macOS-13 instead of latest (the x86 runners).
And then maybe we consider a different imageio plugin with better wheel support: 
https://github.com/napari/napari-animation/issues/217

GenevieveBuckley commented:
> why not use the homebrew installed on the macOS runners?

I saw several brew actions available on the github marketplace, and assumed that meant the default runners did not come with it already installed.

GenevieveBuckley commented:
I'm going to close this - even though I've removed the marketplace github action (and this is change can be seen in the workflow file), the action is still being run on the CI jobs. It's probably easiest to delete this, instead of figuring out if there is a cache somewhere.

GenevieveBuckley commented:
Superseded by https://github.com/napari/napari-animation/pull/222
